src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3272,6 +3270,16 @@ private static bool TryGetNumberValue(string strNum, bool hex, bool real, char s
                             {
                                 result = ((long)Convert.ChangeType(d, typeof(long), CultureInfo.InvariantCulture)) * multiplier;
                             }
+                            if (suffix == 'u' || suffix == 'U') {
+                                ulong testresult = ((ulong)Convert.ChangeType(d, typeof(ulong), CultureInfo.InvariantCulture)) * (ulong)multiplier;
```

Comment:
>Convert.ChangeType [](start = 59, length = 18)

This can raise an exception. The parser and tokenizer try hard to avoid any exception.

Reply:
I simply copied what was already being used in the parser.

Suggested alternatives?

Reply:
This follows the same code path as the original calls, where the double parse is confirmed before the conversion occurs. Shouldn't be an issue. :)

Reply:
So you're saying `-1.0u` doesn't raise an exception?

Reply:
You are conflating 2 things - reporting an error does not require raising an exception.

As I stated in another comment, the intent is to never raise an exception while parsing, but this function does sometimes, so I guess it's fine to do so here too.

The parser *by design* avoids exceptions for performance reasons. Exceptions should be exceptional, and because the parser is used in interactive scenarios, errors are not exceptional.

So yes - report an error, but not via an exception.

Reply:
Gotcha. Yeah, I'm not familiar enough with c# or the internals of how PS handles errors vs exceptions to be clear on that difference at the moment. Thank you!